# General Parameters
use_scan_matching: true
use_scan_barycenter: true
minimum_time_interval: 3600
minimum_travel_distance: 0.2 
minimum_travel_heading: 0.174                  #in radians
scan_buffer_size: 110
scan_buffer_maximum_scan_distance: 50.0
use_scan_range: 50

# Correlation Parameters - Correlation Parameters
# smear_deviation Must be between [0.5 * resolution, 10 * resolution]
correlation_search_space_dimension: 0.3
correlation_search_space_resolution: 0.05
correlation_search_space_smear_deviation: 0.03

# Scan Matcher Parameters
distance_variance_penalty: 0.3              # gets squared later
angle_variance_penalty: 0.1                # in degrees (gets converted to radians then squared)
fine_search_angle_offset: 0.00349               # in degrees (gets converted to radians)
coarse_search_angle_offset: 0.349            # in degrees (gets converted to radians)
coarse_angle_resolution: 0.0349                # in degrees (gets converted to radians)
minimum_angle_penalty: 0.9
minimum_distance_penalty: 0.5
use_response_expansion: true

# back-end
use_back_end: true     # 是否启用后端优化
do_loop_closing: true  # 是否启用回环检测
marker_count: 0
solver_type: spa_solver # spa_solver, g2o_solver, ceres_solver, gtsam_solver

# Correlation Parameters - Loop Closure Parameters
# smear_deviation Must be between [0.5 * resolution, 10 * resolution]
loop_search_space_dimension: 15.0
loop_search_space_resolution: 0.1
loop_search_space_smear_deviation: 0.3

link_match_minimum_response_fine: 0.1
link_scan_maximum_distance: 1.5
loop_search_maximum_distance: 15.0

loop_match_minimum_chain_size: 5
loop_match_maximum_variance_coarse: 3     # gets squared later
loop_match_minimum_response_coarse: 0.35
loop_match_minimum_response_fine: 0.45